pkg/mapstreamer/server.go

```diff
@@ -35,9 +36,16 @@ func (m *server) Start(ctx context.Context) error {
 	ctxWithSignal, stop := signal.NotifyContext(ctx, syscall.SIGINT, syscall.SIGTERM)
 	defer stop()
 
-	// write server info to the file
+	// write server info to the file, we need to add metadata to ensure selection of the
+	// correct map mode, in this case streaming map
+	serverInfo := info.GetDefaultServerInfo()
+	serverInfo.Metadata = map[string]string{info.MapModeMetadata: string(info.StreamMap)}
+	if err := info.Write(serverInfo, info.WithServerInfoFilePath(m.opts.serverInfoFilePath)); err != nil {
+		return err
+	}
+
```

Comment:
I see that you moved this piece of code from `PrepareServer`, now we have two places writing the server file, outside and inside of PrepareServer.

Can we choose one? either all within PrepareServer or none.

Reply:
I prefer moving back to PrepareServer by adding the ServerInfo object to the input parameters of PrepareServer's.